src/main/java/org/jabref/model/entry/Month.java

```diff
@@ -115,6 +117,9 @@
      *         form
      */
 	private static Optional<Month> parseGermanShortMonth(String value) {
+        // support for lowercase German month
+        value = value.substring(0,1).toUpperCase() + value.substring(1);
```

Comment:
Please remove this and the line above. Just use [equalsIgnoreCase](https://www.tutorialspoint.com/java/java_string_equalsignorecase.htm).

Reply:
Now no German months are accepted with lowercase, just `mae`.

Reply:
I overlooked this case. I think, you have to change the if then to following

    if (("Mae".equalsIgnoreCase(value)) || ("Maerz".equalsIgnoreCase(value))) {

Please also ensure that a test case for `Maerz` is in there. If it was already, then I don't get why that code still worked.

Reply:
The dateformatter is case-sensitive by default. (Adapting) the following code should work:
````java
DateTimeFormatter formatter = new DateTimeFormatterBuilder()
    .parseCaseInsensitive()
    .appendPattern("yyyy-MMM")
    .toFormatter(Locale.US);
````